From efa4161ef1384071055ce3f09c67a778cd58b0d7 Mon Sep 17 00:00:00 2001
From: Shitful <2292683261@qq.com>
Date: Sun, 7 Jun 2026 22:12:11 +0800
Subject: [PATCH 5/6] align

---
 bsp/abstract-machine/src/init.c | 27 +++++++++++++++------------
 1 file changed, 15 insertions(+), 12 deletions(-)

diff --git a/bsp/abstract-machine/src/init.c b/bsp/abstract-machine/src/init.c
index 060946433..9bcfd72c3 100644
--- a/bsp/abstract-machine/src/init.c
+++ b/bsp/abstract-machine/src/init.c
@@ -6,27 +6,30 @@
 #define AM_APPS_HEAP_SIZE 0x2000000
 #define RT_HW_HEAP_BEGIN heap.start
 #define RT_HW_HEAP_END heap.end
-
 __attribute__((section(".ssbl"))) static void *
 __ssbl__memcpy(void *out, const void *in, size_t n) {
   unsigned char *d = (unsigned char *)out;
   const unsigned char *s = (const unsigned char *)in;
 
-  while ((((uintptr_t)d) & 0x3) && n) {
-    *d++ = *s++;
-    n--;
+  if ((((uintptr_t)d ^ (uintptr_t)s) & 0x3) == 0) {
+    while ((((uintptr_t)d) & 0x3) && n != 0) {
+      *d++ = *s++;
+      n--;
+    }
+
+    while (n >= 4) {
+      *(uint32_t *)d = *(const uint32_t *)s;
+      d += 4;
+      s += 4;
+      n -= 4;
+    }
   }
 
-  while (n >= 4) {
-    *(uint32_t *)d = *(const uint32_t *)s;
-    d += 4;
-    s += 4;
-    n -= 4;
-  }
-
-  while (n-- > 0) {
+  while (n != 0) {
     *d++ = *s++;
+    n--;
   }
+
   return out;
 }
 
-- 
2.54.0

